wagtail/wagtailsnippets/edit_handlers.py

```diff
@@ -35,6 +49,10 @@ def render_as_field(self):
             'snippet_type_name': self.snippet_type_name,
         }))
 
+    @property
+    def snippet_type_name(self):
+        return force_text(self.target_content_type()._meta.verbose_name)
```

Comment:
Defining `snippet_type_name` as a property doesn't work, as the `widget_overrides` method expects it to exist as an attribute of the class itself. This results in the button text of the chooser widget becoming something like `Choose <property object at 0xb5bebfa4>`. I guess this will need refactoring so that the `target_content_type` logic happens outside of the class...